lib/bolt/transport/ssh.rb

```diff
@@ -115,8 +115,14 @@ def run_script(target, script, arguments, options = {})
       end
 
       def run_task(target, task, arguments, options = {})
-        executable = target.select_impl(task, PROVIDED_FEATURES)
-        raise "No suitable implementation of #{task.name} for #{target.name}" unless executable
+        if task.input_method == 'api'
```

Comment:
Is there a better way to use the file passed on the API than this?

Reply:
According to https://github.com/net-ssh/net-scp it can upload from an in-memory buffer with `scp.upload! StringIO.new("some data to upload"), "/remote/path"`.

Reply:
A temporary file seems like the only short-term solution for WinRM though. We'd have to modify winrm-fs otherwise.

Reply:
This isn't what `task.input_method` is normally used for. Right now it precludes passing the input_method for the task (part of its metadata) via the API.

Reply:
Probably easiest to put the file content in something called `file_content`, then key off its existence (i.e. `task.respond_to? :file_content`.